unified-runtime/test/conformance/exp_command_buffer/enqueue.cpp

```diff
@@ -53,53 +63,108 @@ struct urEnqueueCommandBufferExpTest
         cmd_buf_handle, kernel, n_dimensions, &global_offset, &global_size,
         nullptr, 0, nullptr, 0, nullptr, 0, nullptr, nullptr, nullptr,
         nullptr));
+
+    // Schedule memory copying in order to prolong the buffer execution
+    for (int i = 0; i < num_copy_buffers; i++) {
+      ASSERT_SUCCESS(urCommandBufferAppendUSMMemcpyExp(
+          cmd_buf_handle, dst_buffers[i], src_buffers[i],
+          buffer_size * sizeof(int), 0, nullptr, 0, nullptr, nullptr, nullptr,
+          nullptr));
+    }
+
     ASSERT_SUCCESS(urCommandBufferFinalizeExp(cmd_buf_handle));
   }
 
   virtual void TearDown() override {
-    if (in_order_queue) {
-      EXPECT_SUCCESS(urQueueRelease(in_order_queue));
-    }
-
-    if (out_of_order_queue) {
-      EXPECT_SUCCESS(urQueueRelease(out_of_order_queue));
+    if (in_or_out_of_order_queue) {
+      EXPECT_SUCCESS(urQueueRelease(in_or_out_of_order_queue));
     }
 
     if (device_ptr) {
       EXPECT_SUCCESS(urUSMFree(context, device_ptr));
     }
 
-    UUR_RETURN_ON_FATAL_FAILURE(urCommandBufferExpExecutionTest::TearDown());
+    for (int i = 0; i < num_copy_buffers; i++) {
+      if (dst_buffers[i]) {
+        EXPECT_SUCCESS(urUSMFree(context, dst_buffers[i]));
+      }
+
+      if (src_buffers[i]) {
+        EXPECT_SUCCESS(urUSMFree(context, src_buffers[i]));
+      }
+    }
+
+    UUR_RETURN_ON_FATAL_FAILURE(
+        urCommandBufferExpExecutionTestWithParam::TearDown());
   }
 
-  ur_queue_handle_t in_order_queue = nullptr;
-  ur_queue_handle_t out_of_order_queue = nullptr;
+  ur_queue_handle_t in_or_out_of_order_queue = nullptr;
```

Comment:
just call it queue.

Reply:
Can't we use the one from the parent class then? I think we can just modify queue_properties BEFORE calling parent SetUp (`UUR_RETURN_ON_FATAL_FAILURE(
        urCommandBufferExpExecutionTestWithParam::SetUp());`) to make it in-order or out-of-order